src/components/Modal.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, Fragment } from 'react';
+import { Dialog, Transition } from '@headlessui/react'
+
+type Props = {
+  title: string,
+  isOpen: boolean,
+  setIsOpen: (open: boolean) => void,
+  children: JSX.Element,
+  buttonMsg: string
+}
+
+const Modal: React.FunctionComponent<Props> = (props: Props) => {
+  const { title, children, isOpen, setIsOpen, buttonMsg } = props;
+  console.log('helo')
+
+  return (
+    <Transition appear show={isOpen} as={Fragment}>
+      <Dialog open={isOpen} onClose={() => setIsOpen(false)} as="div" className="fixed inset-0 z-40 overflow-y-auto">
+
+        <div className="min-h-screen px-4 text-center">
+          <Dialog.Overlay className="fixed inset-0 bg-black opacity-30" />
+
+
+          <span
+            className="inline-block h-screen align-middle"
+            aria-hidden="true"
+          >
+            &#8203;
+          </span>
+          <Transition.Child
+            as={Fragment}
+            enter="ease-out duration-300"
+            enterFrom="opacity-0 scale-95"
+            enterTo="opacity-100 scale-100"
+            leave="ease-in duration-200"
+            leaveFrom="opacity-100 scale-100"
+            leaveTo="opacity-0 scale-95"
+          >
+            <div className="inline-block w-full max-w-md p-6 my-8 overflow-hidden text-left align-middle transition-all transform bg-white dark:bg-gray-700 shadow-xl rounded-2xl">
+              <Dialog.Title
+                as="h3"
+                className="text-lg font-medium leading-6 text-gray-900 dark:text-white"
+              >
+                {title}
+              </Dialog.Title>
+              <div className="mt-2">
+                {children}
+              </div>
+
+              <div className="mt-4">
+                <button
+                  type="button"
+                  className="inline-flex justify-center px-4 py-2 text-sm font-medium text-blue-900 bg-blue-100 border border-transparent rounded-md hover:bg-blue-200 focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-blue-500"
```

Comment:
maybe start using @apply ?

Reply:
:(

Reply:
u can do a seperate PR to move a bunch of long css into their own classes